common/sketch/src/main/java/org/apache/spark/util/sketch/CountMinSketchImpl.java

```diff
@@ -348,4 +374,30 @@ public static CountMinSketchImpl readFrom(InputStream in) throws IOException {
 
     return new CountMinSketchImpl(depth, width, totalCount, hashA, table);
   }
+
+  @Override
+  public void writeExternal(ObjectOutput out) throws IOException {
+    try (ByteArrayOutputStream bos = new ByteArrayOutputStream()) {
+      this.writeTo(bos);
+      byte[] bytes = bos.toByteArray();
+      out.writeObject(bytes);
```

Comment:
Oh actually we can just use extend `Serializable` and use private `writeObject` and `readObject` override serialization. There we have `ObjectInputStream`/`ObjectOutputStream` instead of `ObjectInput`/`ObjectOutput`.

cc @cloud-fan